l10n_br_nfe/models/document_line.py

```diff
@@ -344,39 +344,56 @@ def _prepare_import_dict(self, values, model=None):
             'vBC': str("%.02f" % self.icms_base),
             'pRedBC': str("%.04f" % self.icms_reduction),
-            'pICMS': str("%.04f" % self.nfe40_pICMS),
-            'vICMS': str("%.02f" % self.nfe40_vICMS),
+            'pICMS': str("%.04f" % self.icms_percent),
+            'vICMS': str("%.02f" % self.icms_value),
+            'vICMSSubstituto': str("%.02f" % self.icms_substitute),
 
-            # ICMS ST
+            # ICMS SUBSTITUIÇÃO TRIBUTÁRIA
             'modBCST': self.icmsst_base_type,
             'pMVAST': str("%.04f" % self.icmsst_mva_percent),
             'pRedBCST': str("%.04f" % self.icmsst_reduction),
             'vBCST': str("%.02f" % self.icmsst_base),
             'pICMSST': str("%.04f" % self.icmsst_percent),
             'vICMSST': str("%.02f" % self.icmsst_value),
-            # 'pBCOp': '',
             'UFST': self.partner_id.state_id.code,
 
-            # ICMS ST Dest
-            'vBCSTRet': '',
-            'pST': '',
-            'vICMSSubstituto': '',
-            'vICMSSTRet': '',
-            'vBCFCPSTRet': '',
-            'pFCPSTRet': '',
-            'vFCPSTRet': '',
-            'vBCSTDest': '',
-            'vICMSSTDest': '',
-            'pRedBCEfet': '',
-            'vBCEfet': '',
-            'pICMSEfet': '',
-            'vICMSEfet': '',
+            # ICMS COBRADO ANTERIORMENTE POR SUBSTITUIÇÃO TRIBUTÁRIA
+            'vBCSTRet': str("%.02f" % self.icmsst_wh_base),
+            'pST': str("%.04f" % (self.icmsst_wh_percent + self.icmsfcp_wh_percent)),
+            'vICMSSTRet': str("%.02f" % self.icmsst_wh_value),
+            'vBCFCPSTRet': str("%.02f" % self.icmsfcp_base_wh),
+            'pFCPSTRet': str("%.04f" % self.icmsfcp_wh_percent),
+            'vFCPSTRet': str("%.02f" % self.icmsfcp_value_wh),
+            'pRedBCEfet': str("%.04f" % self.icms_effective_reduction),
+            'vBCEfet': str("%.02f" % self.icms_effective_base),
+            'pICMSEfet': str("%.04f" % self.icms_effective_percent),
+            'vICMSEfet': str("%.02f" % self.icms_effective_value),
+
+            # DIFAL ICMS
+            'vBCUFDest': str("%.02f" % self.icms_destination_base),
+            'pFCPUFDest': str("%.04f" % self.icmsfcp_percent),
+            'pICMSUFDest': str("%.04f" % self.icms_origin_percent),
+            'pICMSInter': str("%.04f" % self.icms_destination_percent),
+            'pICMSInterPart': str("%.04f" % self.icms_sharing_percent),
+            'vFCPUFDest': str("%.02f" % self.icmsfcp_value),
+            'vICMSUFDest': str("%.02f" % self.icms_destination_value),
+            'vICMSUFRemet': str("%.02f" % self.icms_origin_value),
+
+            # ICMS SIMPLES NACIONAL
+            'CSOSN': self.icms_cst_id.code,
+            'pCredSN': str('%.04f' % self.icmssn_percent),
+            'vCredICMSSN': str("%.02f" % self.icmssn_credit_value),
+
+            # FUNDO DE COMBATE À POBREZA
+            'vBCFCPST': str('%.02f' % self.icmsfcp_base),
```

Comment:
Semelhante aos campos do DIFAL esses campos não estão dentro das tags do ICMS mas da tag ICMSUFDest

Reply:
@renatonlima mas esses três campos não foram adicionados as tags do ICMS recentemente? Pelo manual do contribuinte de novembro de 2020, eles estão no local correto.

Reply:
Ola @ygcarvalh teria como vc dar um gas nessa colocações https://github.com/OCA/l10n-brazil/pull/1240#issuecomment-849638275 para a gente poder chegar nesse PR pr favor?

Reply:
@rvalyi é claro.

Reply:
valeu pelo esforço @ygcarvalh . Sobre a maioria desses campos eles sao sim da tag ICMSST como pode ser verificado aqui:
https://github.com/OCA/l10n-brazil/blob/12.0/l10n_br_nfe_spec/models/v4_00/leiauteNFe.py#L1733
cc @renatonlima 

Agora, sobre as 3 tags do ICMS SIMPLES NACIONAL: CSOSN, pCredSN, vCredICMSSN eu expliquei aqui porque eu filtrei elas da geraçao do l10n_br_nfe_spec (usando o mesmo sistema de filtro que filtra as tags de assinatura do XML). Mas mesmo que fora do l10n_br_nfe_spec/models/v4_00/leiauteNFe.py, pode ser verificado pelo schema da NFe que estao dentro das tags particulares do ICMSSN* https://github.com/akretion/nfelib/blob/master_gen_v4_00/schemas/nfe/v4_00/leiauteNFe_v4.00.xsd#L3219

Nisso eu accredito que esse mapping so vai funcionar se for feito algo semelhante do que fizemos pras tags ICMS* aqui https://github.com/OCA/l10n-brazil/blob/e34d7e4c0fdd38bf8359f7363cd4234bcadcf581/l10n_br_nfe/models/document_line.py#L395
ou seja, ja que as tags especificas foram filtradas do l10n_br_nfe_spec, tem que criar manualmente a tag ICMS ou entao ICMSSN especifica aqui:
https://github.com/OCA/l10n-brazil/blob/e34d7e4c0fdd38bf8359f7363cd4234bcadcf581/l10n_br_nfe/models/document_line.py#L395
Assim como foi feito pro ICMS 5 linhas antes.
E talvez ate botar o mapping desses 3 campos dentro daquele if do icmssn e nao na funçao especifica dos campos do ICMS.

Reply:
@renatonlima na vdd todas essas tags ICMS*, ICMSN*, ICMSST, ICMSPart são filhas da tag ICMS como pode ser visto aqui:
https://github.com/OCA/l10n-brazil/blob/12.0/l10n_br_nfe_spec/models/v4_00/leiauteNFe.py#L1962

Portanto isso deve funcionar. @ygcarvalh seria bom vc botar https://github.com/OCA/l10n-brazil/pull/1281 verde para a gente ter certeza disso. Mas assim de longe me parece correcto.